### PR TITLE
Contribute liberty build tools integration plugins

### PR DESCRIPTION
**Pull request by rajivsen**
Contribute the liberty build tools integration plugin base plugin: com.ibm.ws.st.liberty.buildplugin.integration

and the maven implementation plugin: 
com.ibm.etools.maven.liberty.integration

### CODE REVIEW

**Comment by eharris369** [comment]
All copyrights need to be updated to the open source copyright.  Copyrights should be added to build.properties, build.xml, plugin.properties and plugin.xml files.

File: dev/com.ibm.etools.maven.liberty.integration/src/com/ibm/etools/maven/liberty/integration/xml/internal/XMLDocumentBuilder.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.xml.sax.SAXException;
+
+import com.ibm.etools.maven.liberty.integration.internal.Trace;
+
+public abstract class XMLDocumentBuilder {
```
Comment (eharris369) [inline dev/com.ibm.etools.maven.liberty.integration/src/com/ibm/etools/maven/liberty/integration/xml/internal/XMLDocumentBuilder.java]: There is a common version of this class too.  Are they the same and can this one be removed?

**Comment by eharris369** [comment]
Can more comments be added to the new interfaces?

File: dev/com.ibm.etools.maven.liberty.integration/src/com/ibm/etools/maven/liberty/integration/xml/internal/XMLDocumentBuilder.java (hunk 1 above)
Comment (rajivsen) [reply dev/com.ibm.etools.maven.liberty.integration/src/com/ibm/etools/maven/liberty/integration/xml/internal/XMLDocumentBuilder.java]: Thanks for catching that. I've removed it.

**Comment by rajivsen** [comment]
I've added copyrights and comments as well.

File: dev/com.ibm.etools.maven.liberty.integration/src/com/ibm/etools/maven/liberty/integration/servertype/internal/LibertyMavenServerBehaviour.java [hunk 2]
```diff
@@ -1,24 +1,27 @@
-/**
- * IBM Confidential
- * OCO Source Materials
- * (C) Copyright IBM Corp. 2016, 2017 All Rights Reserved.
- * The source code for this program is not published or otherwise
- * divested of its trade secrets, irrespective of what has
- * been deposited with the U.S. Copyright Office.
- */
+/*******************************************************************************
+ * Copyright (c) 2017 IBM Corporation and others.
```
Comment (eharris369) [inline dev/com.ibm.etools.maven.liberty.integration/src/com/ibm/etools/maven/liberty/integration/servertype/internal/LibertyMavenServerBehaviour.java]: The dates don't match the old copyright.

**Pull request merged**